java/runtime-common/src/main/java/org/ray/core/Worker.java

```diff
@@ -72,166 +73,102 @@ public static void execute(TaskSpec task, LocalFunctionManager funcs) {
-          + "see JarRewriter for options");
-    }
-    byte[] fid = MethodSwitcher.MethodId.get();//get the identity of function from hook
-    MethodSwitcher.IsRemoteCall.set(false);
-    return fid;
+  private RayObjects actorTaskSubmit(UniqueID taskId,
+      MethodId methodId,
+      int returnCount,
+      boolean multiReturn,
+      Object[] args,
+      RayActor<?> actor) {
+    RayInvocation ri = createRemoteInvocation(methodId, args, actor);
+    RayObjects returnObjs = scheduler.submit(taskId, ri, returnCount + 1, multiReturn);
+    actor.setTaskCursor(returnObjs.pop().getId());
+    return returnObjs;
   }
 
   private RayObjects submit(UniqueID taskId,
-                            byte[] fid,
-                            int returnCount,
-                            boolean multiReturn,
-                            Object[] args) {
+      MethodId methodId,
+      int returnCount,
+      boolean multiReturn,
+      Object[] args) {
     if (taskId == null) {
       taskId = UniqueIdHelper.nextTaskId(-1);
     }
     if (args.length > 0 && args[0].getClass().equals(RayActor.class)) {
-      return actorTaskSubmit(taskId, fid, returnCount, multiReturn, args, (RayActor<?>) args[0]);
+      return actorTaskSubmit(taskId, methodId, returnCount, multiReturn, args,
+          (RayActor<?>) args[0]);
     } else {
-      return taskSubmit(taskId, fid, returnCount, multiReturn, args);
+      return taskSubmit(taskId, methodId, returnCount, multiReturn, args);
     }
   }
 
-  private RayObjects actorTaskSubmit(UniqueID taskId,
-                                     byte[] fid,
-                                     int returnCount,
-                                     boolean multiReturn,
-                                     Object[] args,
-                                     RayActor<?> actor) {
-    RayInvocation ri = new RayInvocation(fid, args, actor);
-    RayObjects returnObjs = scheduler.submit(taskId, ri, returnCount + 1, multiReturn);
-    actor.setTaskCursor(returnObjs.pop().getId());
-    return returnObjs;
-  }
-
-  private RayObjects taskSubmit(UniqueID taskId,
-                                byte[] fid,
-                                int returnCount,
-                                boolean multiReturn,
-                                Object[] args) {
-    RayInvocation ri = new RayInvocation(fid, args);
-    return scheduler.submit(taskId, ri, returnCount, multiReturn);
-  }
 
-  public RayObjects rpcCreateActor(UniqueID taskId, UniqueID createActorId, Callable funcRun,
-                                   int returnCount,
-                                   Object[] args) {
-    byte[] fid = fidFromHook(funcRun);
-    RayInvocation ri = new RayInvocation(fid, new Object[] {});
-    return scheduler.submit(taskId, createActorId, ri, returnCount, false);
+  public RayObjects rpc(UniqueID taskId, Class<?> funcCls, Serializable lambda,
```

Comment:
Change the type of `lambda` to `RayFunc`, so that users can't pass other serializable types.

Reply:
ok